src/stateManager/StateManager.ts

```diff
@@ -1379,37 +1493,52 @@ class StateManager {
     }
 
     private async createStateSnapshot(
+        previousStateSnapshot: StateSnapshot,
```

Comment:
why?
previousStateSnapshot is a function of cooridantes. so why add it as an additional argument?